### PR TITLE
Add Inria datamodule 

### PR DESCRIPTION
**Pull request by ashnair1**
- Add Inria datamodule
- Add tests and update test data

~Prediction utilities
Added `predict.py`. Allows running inference on test sets / directory of images. Requires kornia>0.6.4 for fixed `CombineTensorPatches`.~

### CODE REVIEW

File: torchgeo/datamodules/inria.py
Comment (ashnair1) [inline torchgeo/datamodules/inria.py:60]: This causes the following error while building docs

```
WARNING: py:class reference target not found: kornia.augmentation.container.augment.AugmentationSequential
```

Comment (ashnair1) [inline torchgeo/datamodules/inria.py:60]: This causes the following error while building docs

```
WARNING: py:class reference target not found: kornia.augmentation.container.augment.AugmentationSequential
```

File: tests/datamodules/test_inria.py
Comment (adamjstewart) [inline tests/datamodules/test_inria.py:1]: I'm trying to get rid of the tests in `tests/datamodules` because they aren't actually that useful. Instead, I'm trying to test things directly in `tests/trainers` with real pytorch-lightning trainers.

File: predict.py
Comment (adamjstewart) [inline predict.py:1]: Does this have to be in this PR or can it be in a separate PR?

File: conf/inria.yaml
Comment (adamjstewart) [inline conf/inria.yaml:27]: This is pretty small... Note that `conf/inria.yaml` should contain the optimal hyperparameters while `tests/conf/inria.yaml` should contain the bare minimum to get the tests to run.

File: setup.cfg
Comment (adamjstewart) [inline setup.cfg:34]: Note that this is the latest version of Kornia, released only a couple weeks ago. I'm not opposed to requiring this, but it is quite new.

File: torchgeo/datamodules/inria.py
Comment (adamjstewart) [inline torchgeo/datamodules/inria.py:60]: None of our other datamodules take augmentations as input. I'm not opposed to this, but it's something we should discuss. This also relates to #463

File: torchgeo/datasets/utils.py
Comment (adamjstewart) [inline torchgeo/datasets/utils.py:81]: Why is this needed?

File: predict.py
Comment (ashnair1) [reply predict.py]: Better to separate it out. I've removed it and will open another PR once this is merged.

File: tests/datamodules/test_inria.py
Comment (ashnair1) [reply tests/datamodules/test_inria.py:1]: I've added a test config for the dataset but I would like to keep this test for now. Going to need this in the prediction PR.

File: conf/inria.yaml
Comment (ashnair1) [reply conf/inria.yaml:27]: Added `tests/conf/inria.yaml`

File: torchgeo/datasets/utils.py
Comment (ashnair1) [reply torchgeo/datasets/utils.py]: I can elaborate in the prediction PR but the basic idea is to have a general PredictionDataset that represents a directory of images to run inference on.

File: setup.cfg
Comment (ashnair1) [reply setup.cfg]: Removed

Comment (ashnair1) [reply setup.cfg]: This is actually needed for `compute_padding` a function I recently added to kornia. I could rewrite the function here but the corrected `CombineTensorPatches` will be required for the prediction PR.

File: torchgeo/datamodules/inria.py
Comment (ashnair1) [reply torchgeo/datamodules/inria.py:58]: Sure so should I hold off on this till #463 is resolved?  Got this idea from isaac's [dfc2022](https://github.com/isaaccorley/dfc2022-baseline/blob/master/src/datamodule.py) repo and I personally like it.

That being said, this is currently causing the following error while building docs.
```
WARNING: py:class reference target not found: kornia.augmentation.container.augment.AugmentationSequential
```

Comment (adamjstewart) [reply torchgeo/datamodules/inria.py:58]: Can you try adding https://kornia.readthedocs.io/en/latest/ to `intersphinx_mapping` in `docs/conf.py`?

Comment (ashnair1) [reply torchgeo/datamodules/inria.py:58]: That worked.

Comment (calebrob6) [reply torchgeo/datamodules/inria.py:58]: I really like having augmentations as input to the datamodule

**Review by calebrob6 — DISMISSED** [review]

**Comment by ashnair1** [comment]
Docs are failing due to the recent deprecation of `core/lightning.py` in favor of `core/module.py` in pytorch-lightning. Refer PyTorchLightning/pytorch-lightning#12740.

**Comment by ashnair1** [comment]
@adamjstewart @calebrob6 @isaaccorley 

Anything left to address in this PR?

File: conf/inria.yaml
Comment (adamjstewart) [reply conf/inria.yaml:27]: Now that the small values required to make the tests run fast are in `tests/conf/inria.yaml`, should these values be increased? I guess it requires a hyperparam tuning to determine what the best hyperparams are.

File: setup.cfg
Comment (adamjstewart) [inline setup.cfg:34]:
```suggestion
    # kornia 0.6.4+ required for kornia.contrib.compute_padding
```

File: torchgeo/datamodules/inria.py
Comment (adamjstewart) [inline torchgeo/datamodules/inria.py:14]: We should probably avoid using undocumented private methods of external libs. Can you use `..samplers.utils._to_tuple` instead?

File: torchgeo/datamodules/inria.py
Comment (adamjstewart) [inline torchgeo/datamodules/inria.py:19]:
```suggestion
from .utils import dataset_split
from ..datasets import InriaAerialImageLabeling
```
For stuff in `torchgeo/`, we should always use local imports to avoid confusion if someone puts a file or directory in their `PYTHONPATH` or current directory with a similar name. For stuff in `tests/`, we should always use `from torchgeo` imports so we can test the installed copy instead of the local copy. Btw, can you also change this in `datasets/inria.py`? That's the only other place that makes this mistake.

File: torchgeo/datamodules/inria.py
Comment (adamjstewart) [inline torchgeo/datamodules/inria.py:115]: I assume there's a reason we do it despite it being squeezed out later? Might want to elaborate on that

File: torchgeo/datamodules/inria.py
Comment (adamjstewart) [inline torchgeo/datamodules/inria.py:227]: In the other modules, we use something like:
```python
        if (                                                                             
            hasattr(self, "trainer")                                                     
            and self.trainer is not None                                                 
            and hasattr(self.trainer, "training")                                        
            and self.trainer.training                                                    
        ):              
```
to avoid mypy errors. Mypy is actually correct in this case, all of these methods may or may not be None, so it's good to check first whether they are defined.

**Review by adamjstewart — COMMENTED** [review]
Thanks, I think this is really close, just a few minor comments!

Also, I want to apologize for taking so long to review this. I just finished all of my final projects and exams for the semester and started an internship, so it's been a really busy month. I'm hoping to spend more of my weekends on TorchGeo over the summer so hopefully I'll be more responsive from now on!

File: torchgeo/datamodules/inria.py
Comment (ashnair1) [reply torchgeo/datamodules/inria.py]: Done

File: torchgeo/datamodules/inria.py
Comment (ashnair1) [reply torchgeo/datamodules/inria.py]: Done

File: torchgeo/datamodules/inria.py
Comment (ashnair1) [reply torchgeo/datamodules/inria.py]: Makes sense

**Comment by ashnair1** [comment]
> Thanks, I think this is really close, just a few minor comments!
> 
> Also, I want to apologize for taking so long to review this. I just finished all of my final projects and exams for the semester and started an internship, so it's been a really busy month. I'm hoping to spend more of my weekends on TorchGeo over the summer so hopefully I'll be more responsive from now on!

No worries. You're doing plenty as is. Thanks.

Got back to this PR because a colleague wanted to try out a building detector on their own data. Honestly my original motivation was the same. Train a model on this dataset and add utilities for running inference on geotiffs of arbitrary size by leveraging kornia. Got sidetracked by other projects but I'm hoping to get this done soon.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
